src/Features/CSharp/Portable/EmbeddedLanguages/EmbeddedLanguageUtilities.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Features.EmbeddedLanguages
+{
+    internal static class EmbeddedLanguageUtilities
+    {
+        public static string EscapeText(string text, SyntaxToken token)
+        {
+            // This function is called when Completion needs to escape something its going to
+            // insert into the user's string token.  This means that we only have to escape
+            // things that completion could insert.  In this case, the only regex character
+            // that is relevant is the \ character, and it's only relevant if we insert into
+            // a normal string and not a verbatim string.  There are no other regex characters
+            // that completion will produce that need any escaping. 
+            Debug.Assert(token.Kind() == SyntaxKind.StringLiteralToken);
```

Comment:
this is only called throuh the regex completion code-path.  And regex-completion only works on normal strings.  We don't have regex features on interpolated strings because we don't know how the interpolatin parts will affect the regex.  i.e.   `"foo[${...}bar"`.  No clue what's in that `${...}` section, so we can't do regex.

Reply:
@CyrusNajmabadi I assumed as much. Do we have a negative test somewhere in the system for it? I'm imagining it's not in the regex specific code but just the generic embedded language code?

(after the other bug I'm now paranoid about those things...)